app/src/main/java/com/okta/appauth/android/example/TestLoginActivity.java

```diff
@@ -0,0 +1,191 @@
+    private AuthorizationResponse mResponse;
+    private TokenResponse mToken;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        Log.d(TAG, "onCreate");
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.test_activity_login);
+        mButton = findViewById(R.id.start_button);
+        mButton.setOnClickListener(v -> {
+                    mOktAuth.startAuthorization();
+                    //testing config change.
+                    //setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE);
+                }
+        );
+        mTvStatus = findViewById(R.id.status);
+
+        mOktaAccount = new OktaAuthAccount.Builder()
+                .clientId("0oaiv94wtjW7DHvvj0h7")
+                .redirectUri("com.okta.appauth.android.example:/callback")
+                .endSessionRedirectUri("com.okta.appauth.android.example:/logout")
+                .scopes("openid", "profile", "offline_access")
+                .discoveryUri("https://dev-486177.oktapreview.com/oauth2/default")
+                .create();
+
+        mOktaAccountWithRes = new OktaAuthAccount.Builder().withResId(this, R.raw.okta_app_auth_config).create();
+
+        mOktAuth = new OktaAuthManager.Builder(this).withCallback(new AuthorizationCallback() {
+            @Override
+            public void onSuccess(OktaClientAPI clientAPI) {
+                Log.d("TestLoginActivity", "SUCCESS");
+                mClient = clientAPI;
+                mTvStatus.setText("authentication success");
+                mButton.setText("Get profile");
+                mButton.setOnClickListener(v -> getProfile());
+            }
+
+            @Override
+            public void onStatus(String status) {
+                Log.d("TestLoginActivity", status);
+                mTvStatus.setText(status);
+            }
+
+            @Override
+            public void onCancel() {
+                Log.d("TestLoginActivity", "CANCELED!");
+                mTvStatus.setText("canceled");
+            }
+
+            @Override
+            public void onError(String msg, AuthorizationException error) {
+                Log.d("TestLoginActivity", error.errorDescription + "onError" + Thread.currentThread().toString());
+                if (error != null) {
+                    mTvStatus.setText(error.errorDescription);
+                } else {
+                    mTvStatus.setText(msg);
+                }
+            }
+        }).withAccount(mOktaAccount)
+                .withTabColor(getColorCompat(R.color.colorPrimary))
+                .create();
+
+    }
+
+    @Override
+    protected void onStart() {
+        Log.d(TAG, "onStart");
+        super.onStart();
+        if (getIntent().getBooleanExtra(EXTRA_FAILED, false)) {
+            showSnackbar(getString(R.string.auth_canceled));
+        }
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
```

Comment:
I think we can do it without ActivityResult, I am working on POC, I will create a separate branch

Reply:
Great look forward to it. Would like to eliminate the callback. I looked at a few ways but it didn't work if the activity is killed.